library/std/src/sys/pal/vexos/mod.rs

```diff
@@ -0,0 +1,128 @@
+pub mod alloc;
+#[path = "../unsupported/args.rs"]
+pub mod args;
+pub mod env;
+pub mod fs;
+#[path = "../unsupported/io.rs"]
+pub mod io;
+#[path = "../unsupported/net.rs"]
+pub mod net;
+#[path = "../unsupported/os.rs"]
+pub mod os;
+#[path = "../unsupported/pipe.rs"]
+pub mod pipe;
+#[path = "../unsupported/process.rs"]
+pub mod process;
+pub mod stdio;
+#[path = "../unsupported/thread.rs"]
+pub mod thread;
+pub mod time;
+
+use crate::arch::asm;
+use crate::hash::{DefaultHasher, Hasher};
+use crate::ptr::{self, addr_of_mut};
+use crate::time::{Duration, Instant};
+
+#[cfg(not(test))]
+#[no_mangle]
+#[link_section = ".boot"]
+pub unsafe extern "C" fn _start() -> ! {
+    extern "C" {
+        static mut __bss_start: u8;
+        static mut __bss_end: u8;
+
+        fn main() -> i32;
+    }
+
+    // Setup the stack
+    asm!("ldr sp, =__stack_top", options(nostack));
+
+    // VEXos doesn't explicitly clean out .bss, so as a sanity
+    // check we'll fill it with zeroes.
+    ptr::slice_from_raw_parts_mut(
```

Comment:
This comment is just untrue, no? It's not a sanity check, it actually needs to be there for some yet-unknown reason, right?

Reply:
I actually can't recall. I think we do it because PROS also does this, but I can't remember if it explicitly causes issues.